### PR TITLE
Problems pane

### PR DESCRIPTION
**Pull request by tptee**
Adds a problems pane containing dupes and version skews for each bundle.

- Move the inspectpack daemon and messages to the plugin so all frontends can consume them
- Factor out some shared scroll box styles
- Use `blessed.listbar` to implement a tabbed-per=bundle UI for modules and problems. Can navigate with either mouse or corresponding number keys

TODO:
- Find a way to make the `listbar` keyboard shortcut colors consistent with the text
- Fix minor layout issues

### CODE REVIEW

File: dashboard/index.js [hunk 1]
```diff
@@ -1,36 +1,33 @@
 "use strict";
 
-const path = require("path");
-const fs = require("fs");
-const os = require("os");
-
+const _ = require("lodash/fp");
```
Comment (kenwheeler) [inline dashboard/index.js]: thug life

File: dashboard/index.js [hunk 2]
```diff
@@ -194,12 +237,62 @@ class Dashboard {
     this.logText.setContent("");
   }
 
+  layoutProblems() {
```
Comment (kenwheeler) [inline dashboard/index.js]: this should be the name of the blessed constructor, amirite

File: demo/index.js [hunk 3]
```diff
@@ -1,7 +0,0 @@
-import _ from 'lodash';
-
-function stuff() {
-  console.log('things');
-}
-console.log(_.capitalize('oh hai mark'));
```
Comment (kenwheeler) [inline demo/index.js]: wait, mark who

File: plugin/index.js [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
 const DEFAULT_PORT = 9838;
 const ONE_SECOND = 1000;
+const INSPECTPACK_INDEPENDENT_ACTIONS = ["sizes"];
+const INSPECTPACK_PROBLEM_ACTIONS = ["versions", "duplicates"];
+const INSPECTPACK_PROBLEM_TYPE = "problems";
+
+const cacheDir = path.resolve(
+  os.homedir(),
+  ".webpack-dashboard-cache"
+);
```
Comment (kenwheeler) [inline plugin/index.js]: This should be fine permissions wise right?

File: plugin/index.js [hunk 5]
```diff
@@ -22,6 +34,70 @@ function getTimeMessage(timer) {
...
+        format: "object",
+        minified: true,
+        gzip: true
+      })
+        .then(metrics => ({
+          path: bundle.path,
+          metrics
+        }))
+    ))
+      .then(bundle => handler([{
```
Comment (kenwheeler) [inline plugin/index.js]: this notifies the consuming ui's right?

**Comment by kenwheeler** [comment]
really really well done. 🎉

File: demo/index.js (hunk 3 above)
Comment (tptee) [inline demo/index.js]: https://www.youtube.com/watch?v=zLhoDB-ORLQ

File: plugin/index.js (hunk 4 above)
Comment (tptee) [inline plugin/index.js]: Yep, and if it isn't, `inspectpack` falls back to in-memory cache only

File: plugin/index.js (hunk 5 above)
Comment (tptee) [inline plugin/index.js]: Yessir!

File: demo/index.js (hunk 3 above)
Comment (kenwheeler) [inline demo/index.js]: what kind of rabbit hole did you just lead me down

File: plugin/index.js (hunk 4 above)
Comment (kenwheeler) [inline plugin/index.js]: superb

File: demo/index.js (hunk 3 above)
Comment (tptee) [inline demo/index.js]: https://www.youtube.com/watch?v=aNv-tH0Vmik

Comment (tptee) [inline demo/index.js]: 🐇 🕳

**Pull request merged**